erigon-lib/recsplit/eliasfano32/elias_fano.go

```diff
@@ -297,74 +333,105 @@ type EliasFanoIter struct {
 	lowerBitsMask uint64
 	l             uint64
 	upperStep     uint64
+	reverse       bool
 
 	//fields of current value
 	upper    uint64
 	upperIdx uint64
 
 	//fields of next value
-	idx       uint64
 	lowerIdx  uint64
 	upperMask uint64
+
+	itemsIterated uint64
 }
 
 func (efi *EliasFanoIter) Close() {}
 func (efi *EliasFanoIter) HasNext() bool {
-	return efi.idx <= efi.count
+	return efi.itemsIterated <= efi.count
```

Comment:
If we start by `Seek()` (not from beginning) then iterated amount of values will be < total amount of values in EF. I mean If in future we will decide add more iter constructors - for example. Most of high-level iterators now have constuctor: `.Range(from, to, order.Desc, limit)` half-open interval `[from, to)`